packages/flutter_tools/lib/src/native_assets.dart

```diff
@@ -214,45 +214,45 @@ BuildMode nativeAssetsBuildMode(build_info.BuildMode buildMode) {
 ///
 /// Native asset builds cannot be run without a package config. If there is
 /// no package config, leave a logging trace about that.
-Future<bool> hasNoPackageConfig(NativeAssetsBuildRunner buildRunner) async {
+Future<bool> _hasNoPackageConfig(NativeAssetsBuildRunner buildRunner) async {
   final bool packageConfigExists = await buildRunner.hasPackageConfig();
   if (!packageConfigExists) {
     globals.logger.printTrace('No package config found. Skipping native assets compilation.');
   }
   return !packageConfigExists;
 }
 
-/// Checks that if native assets is disabled, none of the dependencies declare
-/// native assets.
-///
-/// If any of the dependencies have native assets, but native assets are
-/// disabled, exits the tool.
-Future<bool> isDisabledAndNoNativeAssets(NativeAssetsBuildRunner buildRunner) async {
-  if (featureFlags.isNativeAssetsEnabled) {
+Future<bool> nativeBuildRequired(NativeAssetsBuildRunner buildRunner) async {
+  if (await _hasNoPackageConfig(buildRunner)) {
     return false;
   }
   final List<Package> packagesWithNativeAssets = await buildRunner.packagesWithNativeAssets();
   if (packagesWithNativeAssets.isEmpty) {
-    return true;
+    return false;
   }
-  final String packageNames = packagesWithNativeAssets.map((Package p) => p.name).join(' ');
-  throwToolExit(
-    'Package(s) $packageNames require the native assets feature to be enabled. '
-    'Enable using `flutter config --enable-native-assets`.',
-  );
+
+  if (!featureFlags.isNativeAssetsEnabled) {
+    final String packageNames = packagesWithNativeAssets.map((Package p) => p.name).join(' ');
+    throwToolExit(
+      'Package(s) $packageNames require the native assets feature to be enabled. '
+      'Enable using `flutter config --enable-native-assets`.',
+    );
+  }
+  return true;
 }
 
 /// Ensures that either this project has no native assets, or that native assets
 /// are supported on that operating system.
 ///
 /// Exits the tool if the above condition is not satisfied.
+// TODO(fujino): this does not appear to do what it says it does.
```

Comment:
wdym?

This test checks that either (A) there are no native assets at all, or (B) it throws that the OS is not supported. This function is only called from call sites where the OS is not supported yet. (And (C) it skips the check if there is no package config.)